sphinx/builders/gettext.py

```diff
@@ -32,7 +32,7 @@ class Message:
     """An entry of translatable message."""
     def __init__(self, text: str, locations: List[Tuple[str, int]], uuids: List[str]):
         self.text = text
-        self.locations = locations
+        self.locations = list(set(locations))
```

Comment:
I'd use `sorted` here -- `set` doesn't guarantee any sort order at a language level.

```suggestion
        self.locations = sorted({*locations})
```